src/controllers/users.js

```diff
@@ -0,0 +1,34 @@
+
+import models from '../models';
+import {
+  status, messages, hashPassword, generateUserToken,
+  successResponse, errorResponse, comparePassword, conflictResponse
+} from '../utils/index';
+
+const registerUser = async (req, res) => {
+  try {
+    const { email } = req.body;
+    const userExits = await models.user.findOne({ where: { email } });
+
+    if (userExits) {
+      return conflictResponse(res, status.conflict, messages.signUp.conflict);
+    }
+
+    req.body.password = await hashPassword(req.body.password);
+    const user = await models.user.create(req.body);
+    const response = user.toJSON();
+
+    delete response.password;
+
+    const token = generateUserToken(user.email, user.id, user.is_admin, user.first_name, user.last_name);
```

Comment:
@oyewoas You are still sending `first_name` and `last_name` to the `generateUserToken` function here. Those values will carry `undefined` from the database.

Also, can we just send the user payload here, and do a destructuring of the parameters at the function definition? More like here we write `const token = generateUserToken(user)` and at the function definition, we write something like `const generateUserToken = ({ email, id, is_admin }) => ...` That way we write less code and make it easier for whoever is using our function.

Please help look into this.

Reply:
ok thanks for this, will look into it